### ISSUE TITLE
Join soft deletes

### ISSUE THREAD

**Issue by PunyFlash** [kirschbaum-development/eloquent-power-joins#22]
As said in the docs, when joining any models which use the SoftDeletes trait, the `"deleted_at" is null` condition added to any joins.
But is there a way to join without this condition? Like `withTrashed()` model method?

**Comment by luisdalmolin** [comment]
Hmm 🤔 

It's definitely something that can be added. A workaround you can use, for now, could be something like this:

```php
$query = UserProfile::joinRelationship('user', function ($join) {
    $join->wheres = collect($join->wheres)->reject(function ($where) {
        return $where['type'] === 'Null'
               && Arr::get($where, 'column') === 'users.deleted_at';
        })->toArray();
    });
});
```

In this case, it's removing the deleted_at condition from the user relationship with the user profile. I'll keep this issue open for implementing this in an easier way.

**Comment by luisdalmolin** [comment]
Actually, this is already in issue #5, so I'm closing this one. Feel free to reopen if I missed anything.

**Comment by luisdalmolin** [comment]
@PunyFlash This is now implemented in version 2.1.0: https://github.com/kirschbaum-development/eloquent-power-joins#soft-deletes
